nomenclature/processor/region.py

```diff
@@ -599,6 +603,18 @@ def _apply_region_processing(
         # cast processed timeseries data and meta indicators to IamDataFrame
         return IamDataFrame(_data, meta=model_df.meta), difference
 
+    def revert(self, df: pyam.IamDataFrame) -> pyam.IamDataFrame:
+        model_dfs = []
+        for model in df.model:
+            model_df = df.filter(model=model)
+            if mapping := self.mappings.get(model):
+                # remove common regions
+                model_df = model_df.filter(
+                    region=mapping.common_region_names, keep=False
+                ).rename(region=mapping.reverse_rename_mapping)
```

Comment:
Deja-vu
```suggestion
                model_df = (
                    model_df
                    .filter(region=mapping.common_region_names, keep=False)
                    .rename(region=mapping.reverse_rename_mapping)
                )
```

Reply:
Ah yes just double checked black does not like that one and if I change it, it reverts it back.

Reply:
🤯